### DISCUSSION TITLE
The minimum amount of GPU memory required to train the RT-DETR model

### DISCUSSION THREAD

**Discussion by Thiraphat-K** [ultralytics/ultralytics#4977] [Q&A]
@glenn-jocher Hello, I have a question.
Although I have read in the ultralytics document that you should use at least 8 GB of GPU memory, I don't know what the minimum requirement is for RT-DETR. I tried using Google Colab once with `epochs=150` without any additional arguments, and I was able to train the model without any problems. However, when I tried to increase the batch size to > 16 and `amp=False`, I got a CUDA out of memory error. I then tried to reduce the batch size to < 16, but I still got the same error.

Thank you for your help. :pray::pray::pray:

**Comment by ExtReMLapin** [comment]
**I kept AMP enabled but divided the lr by 10.**
My batch size on a 24gb gpu (rtx 3090 (ti?)) was very low, from memory because of the used 2048px resolution.

**Reply by Thiraphat-K** [reply]
@ExtReMLapin Google Colab uses a Tesla T4 GPU with only 15 GB of memory (but in reality, it is not fully usable). I want to use a batch size of no more than 50. So, based on what you said, I cannot use the batch size I want, right?

**Reply by ExtReMLapin** [reply]
It depends of input image size and model size.
2048px wide uses 4 time more vram than 1024 ones, which is even more than default ~640px images.
So it really depends, give it a try, start low and go up

**Reply by ExtReMLapin** [reply]
By low I mean try at 1

**Reply by Thiraphat-K** [reply]
@ExtReMLapin  My input size is 300x150 pixels, and I use a batch size of 16 (the default value for the model). I use this with the rt-detr model of size L, and I don't have any problems.

**Comment by Thiraphat-K** [comment]
@glenn-jocher Help me pls 🙏. I'm having some doubts about running a model on a Jetson Nano board. Even though I used the `.to(device)` command, the GPU usage didn't change at all. I've read that the minimum recommended memory size is 8 GB, but the Jetson only has 4 GB. Does this mean that even if I call it, it won't actually work?

**Reply by ExtReMLapin** [reply]
you gonna need a bigger gpu if you wanna train

**Reply by Thiraphat-K** [reply]
@ExtReMLapin  I didn't train it, I just used predict.

**Reply by ExtReMLapin** [reply]
You can predict on CPU if it doesn't fit in GPU, holding the model in VRAM uses a non compressible vram count, and the rest depends of image resolution.

**Reply by Thiraphat-K** [reply]
@ExtReMLapin Thank you very much for your help 🙏🙏. Let me summarize that if I want to predict on a Jetson Nano board with GPU: 128-Core Maxwell and Memory: 4GB, which I have tested again and found that it uses the GPU. However, some models like RT-DETR cannot be done on the Jetson because it is too heavy.

**Reply by ExtReMLapin** [reply]
Another way would be to quantize this model, there are tools over there and there, not sure if it's possible with RT-DETR

**Reply by Thiraphat-K** [reply]
@ExtReMLapin Do you have any recommendations for a tool that I can use?
